tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -540,3 +542,31 @@ def test_vmgenid(guest_kernel_linux_6_1, rootfs, microvm_factory, snapshot_type)
 
         # Update the base for next iteration
         base_snapshot = snapshot
+
+
+def test_snapshot_rename_interface(uvm_nano, microvm_factory):
+    """
+    Test that we can restore a snapshot and point its interface to a
+    different host interface.
+    """
+    base_iface = net_tools.NetIfaceConfig.with_id(0)
+
+    vm = uvm_nano
+    iface1 = dataclasses.replace(base_iface, tap_name="tap1")
+    vm.add_net_iface(iface=iface1)
+    # Create an interface but don't attach it to the device
+    vm.start()
+    vm.wait_for_up()
+
+    snapshot = vm.snapshot_full()
+
+    restored_vm = microvm_factory.build()
+    restored_vm.spawn()
+    iface2 = dataclasses.replace(base_iface, tap_name="tap2")
+    snapshot.net_ifaces.clear()
+    snapshot.net_ifaces.append(iface2)
```

Comment:
I don't follow this logic very well. Why do we need to do that? (Not saying it's wrong, maybe it's just me not understanding very well what the snapshot object does exactly.)